L1Trigger/DTTriggerPhase2/interface/MuonPathAnalyzerInChamber.h

```diff
@@ -0,0 +1,141 @@
+#ifndef L1Trigger_DTTriggerPhase2_MuonPathAnalyzerInChamber_cc
+#define L1Trigger_DTTriggerPhase2_MuonPathAnalyzerInChamber_cc
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/MuonDetId/interface/DTChamberId.h"
+#include "DataFormats/MuonDetId/interface/DTSuperLayerId.h"
+#include "DataFormats/MuonDetId/interface/DTLayerId.h"
+#include "DataFormats/MuonDetId/interface/DTWireId.h"
+#include "DataFormats/DTDigi/interface/DTDigiCollection.h"
+
+#include "L1Trigger/DTTriggerPhase2/interface/muonpath.h"
+#include "L1Trigger/DTTriggerPhase2/interface/analtypedefs.h"
+#include "L1Trigger/DTTriggerPhase2/interface/constants.h"
+#include "L1Trigger/DTTriggerPhase2/interface/MuonPathAnalyzer.h"
+
+#include "CalibMuon/DTDigiSync/interface/DTTTrigBaseSync.h"
+#include "CalibMuon/DTDigiSync/interface/DTTTrigSyncFactory.h"
+
+#include "L1Trigger/DTSectorCollector/interface/DTSectCollPhSegm.h"
+#include "L1Trigger/DTSectorCollector/interface/DTSectCollThSegm.h"
+
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "Geometry/DTGeometry/interface/DTGeometry.h"
+#include "Geometry/DTGeometry/interface/DTLayer.h"
+
+#include <iostream>
+#include <fstream>
+
+// ===============================================================================
+// Previous definitions and declarations
+// ===============================================================================
+const int NLayers = 8;
+typedef std::array<LATERAL_CASES, NLayers> TLateralities;
```

Comment:
In principle `NLayers` should be `constexpr` to signify that it is a compile-time constant. The `NLayers` is too generic name to be in global namespace. By quick look it seems to me that both of these could be moved inside the `MuonPathAnalyzerInChamber` class definition.